ap/production/normalization.py

```diff
@@ -0,0 +1,72 @@
+# coding: utf-8
+
+"""
+Column production methods related to sample normalization event weights.
+"""
+
+import law
+import order as od
+
+from ap.production import Producer, producer
+from ap.util import maybe_import
+from ap.columnar_util import set_ak_column
+
+ak = maybe_import("awkward")
+
+
+@producer(
+    uses={"LHEWeight.originalXWGTUP"},
+    produces={"normalization_weight"},
+)
+def normalization_weights(
+    events: ak.Array,
+    selection_stats: dict,
+    config_inst: od.Config,
+    dataset_inst: od.Dataset,
+    **kwargs,
+) -> ak.Array:
+    """
+    Uses luminosity information of *config_inst*, the cross section of a process obtained through
+    *dataset_inst* and the sum of event weights from *selection_stats* to assign each event a
+    normalization weight.
+    """
+    # get lumi and cross section of the first process contained in the dataset
+    # TODO: the process should depend on the process id per event for which we need a LUT
+    #       that is cached through a simple mutable argument
+    lumi = config_inst.x.luminosity.nominal
+    xs = dataset_inst.processes.get_first().get_xsec(config_inst.campaign.ecm).nominal
+
+    # compute the weight and store it
+    norm_weight = events.LHEWeight.originalXWGTUP * lumi * xs / selection_stats["sum_mc_weight"]
+    set_ak_column(events, "normalization_weight", norm_weight)
+
+    return events
+
+
+@normalization_weights.requires
+def normalization_weights_requires(self: Producer, task: law.Task, reqs: dict) -> dict:
```

Comment:
Some comment as above regarding the type hinting of the contents of reqs and the return dict

Reply:
The requirements dict is fully arbitrary as is the return value, so I wouldn't assume and hint any structure here.